_posts/2017/2017-12-31-avoid-double-negatives.md

```diff
@@ -0,0 +1,66 @@
+---
+title: 'Avoid Double Negatives in Conditionals'
+description: 'if unless else end'
+layout: article
+category: ruby
+image:
+  base: '2017/avoid-double-negatives'
+  alt: 'Two directions sign'
+  source: 'https://unsplash.com/photos/lPQIndZz8Mo'
+  credit: 'Pablo Garcia Saldaña'
+---
+
+One of Ruby’s strengths is its delightful, built-in syntactic sugar. One example of this is the keyword `unless`, which you can use in place of using `if` combined with a negative statement.
+
+However, given the flexibility of Ruby's syntax, it is easy to make code harder to understand than it needs to be.
+
+## Instead of…
+
+...overcomplicating your conditionals.
+
+```ruby
+# Example 1
+unless something.nil?
+  # do something
+end
+
+# Example 2
+if !something.nil?
+  # do something
+end
+
+# Example 3
+if !!something
+  # do something
+end
+```
+
+
+## Use…
+
+```ruby
+# Instead of Examples 1,2 & 3
+if something
+  # do something
+end
+```
+
+
+## But why?
+
+Ruby’s conditional syntax is ‘truthy’, meaning that anything that is `nil` is also considered to be `false` and anything not-`nil` can be considered to be `true`.
+
+Performing a `#nil?` check in a negative conditional, as in the first two examples, is often redundant as any non-`nil` value is `true`. So you can reverse the conditional and remove the check for `nil` and end up with clearer code.
+
+The syntax of `!!`, in the third example, is a shorthand for turning any value (either ‘truthy’ or ‘falsey’) into the _actual boolean values_ `true` or `false`. However, given Ruby's ‘truthy’ conditionals you don’t need to perform this redundant conversion.
+
+Avoid using an `else` block with an `unless`. This structure is hard to reason about because the `else` block acts as a double negative of the condition. Instead, replace the `unless` for a logically equivalent `if` and swap the blocks of code in the two branches of the conditional around.
```

Comment:
*the `else` block 'has to be interpreted' as a double negative.. or 'reads as a' double negative.

Not sure what to say here but "acts as a double negative of the condition" isn't clear.

Reply:
I would get rid of 'logically equivalent' and explain what to do:
"Instead, replace the `unless` for `if` and swap around the blocks of code in the two branches of the conditional." (Note how I moved the word 'around' to earlier in the sentence...I think that's the correct way with respect to grammar, but not sure. Either way, I think it makes that sentence easier to read).